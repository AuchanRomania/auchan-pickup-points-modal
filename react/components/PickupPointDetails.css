/* main structure */

.modalDetails {
  display: flex;
  flex-direction: column;
  z-index: 101;
  flex: 1 1 auto;
  position: relative;
}

.modalDetailsTop {
  flex: none;
  margin-bottom: 1px;
}

.modalDetailsMiddle {
  overflow: auto;
  flex: 1 1 auto;
  min-height: 0; /* fix firefox overflow bug */
  -webkit-overflow-scrolling: touch;
}

.modalDetailsBottom {
  background: white;
  flex: none;
  bottom: 0;
  padding: 10px;
}

/* middle section */

.modalDetailsStore {
  padding: 12px;
}

.modalDetailsInfo {
  padding: 15px;
}

.modalDetailsGroup {
  margin-bottom: 20px;
}

.modalDetailsInfoTitle {
  font-weight: 300;
  margin-top: 0;
}

/* business hours */

.modalDetailsHours {
  font-size: 12px;
  width: 100%;
}

.modalDetailsHours td {
  border-bottom: 1px solid #f2f4f5;
  padding-top: 6px;
}

.modalDetailsHoursRange,
.modalDetailsHoursClosed {
  text-align: right;
}

.modalDetailsHoursClosed {
  color: #ff4c4c;
}

/* links and icons */

.modalDetailsBackLnk {
  display: block;
  text-align: left;
  padding: 8px 12px 9px;
}

.iconBackPickupPointsList {
  font-size: 25px;
  vertical-align: sub;
  margin-right: 12px;
  text-decoration: none;
}
