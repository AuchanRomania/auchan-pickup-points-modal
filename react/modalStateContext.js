import React from 'react'

export const ModalStateContext = React.createContext()

export function injectState(Component) {
  return function StateInjectedComponent(props) {
    return (
      <ModalStateContext.Consumer>
        {({
          activeState,
          activeSidebarState,
          geolocationStatus,
          lastState,
          lastSidebarState,
          lastMapCenterLatLng,
          selectedPickupPoint,
          setGeolocationStatus,
          setActiveState,
          setActiveSidebarState,
          setAskForGeolocation,
          setSelectedPickupPoint,
          setMapCenterLatLng,
          shouldSearchArea,
          setShouldSearchArea,
        }) => (
          <Component
            {...props}
            activeState={activeState}
            activeSidebarState={activeSidebarState}
            geolocationStatus={geolocationStatus}
            lastState={lastState}
            lastSidebarState={lastSidebarState}
            lastMapCenterLatLng={lastMapCenterLatLng}
            selectedPickupPoint={selectedPickupPoint}
            setGeolocationStatus={setGeolocationStatus}
            setActiveState={setActiveState}
            setActiveSidebarState={setActiveSidebarState}
            setAskForGeolocation={setAskForGeolocation}
            setMapCenterLatLng={setMapCenterLatLng}
            setSelectedPickupPoint={setSelectedPickupPoint}
            shouldSearchArea={shouldSearchArea}
            setShouldSearchArea={setShouldSearchArea}
          />
        )}
      </ModalStateContext.Consumer>
    )
  }
}
