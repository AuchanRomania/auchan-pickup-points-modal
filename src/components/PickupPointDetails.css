/* main structure */

.pkpmodal-pickup-point-details {
  display: flex;
  flex-direction: column;
  z-index: 101;
  flex: 1 1 auto;
  position: relative;
}

.pkpmodal-pickup-point-details-top {
  flex: none;
}

.pkpmodal-pickup-point-details-middle {
  overflow: auto;
  flex: 1 1 auto;
}

.pkpmodal-pickup-point-details-bottom {
  background: white;
  flex: none;
  bottom: 0;
  padding: 10px;
}

/* middle section */

.pkpmodal-pickup-point-details-store {
  padding: 12px;
}

.pkpmodal-pickup-point-details-info {
  padding: 15px;
}

.pkpmodal-pickup-point-details-group {
  margin-bottom: 20px;
}

.pkpmodal-pickup-point-details-info-title {
  margin-top: 0;
}

/* links and icons */

.pkpmodal-link-back-pickup-points-list {
  display: block;
  text-align: left;
  padding: 8px 12px 9px;
}

.pkpmodal-icon-back-pickup-points-list {
  font-size: 25px;
  vertical-align: sub;
  margin-right: 12px;
  text-decoration: none;
}
