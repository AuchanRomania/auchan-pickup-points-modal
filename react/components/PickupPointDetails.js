import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { injectIntl, intlShape } from 'react-intl'
import { translate } from '../utils/i18nUtils'

import {
  getUnavailableItemsByPickup,
  getItemsWithPickupPoint,
  formatBusinessHoursList,
} from '../utils/pickupUtils'

import PickupPoint from './PickupPoint'
import ProductItems from './ProductItems'
import PickupPointHour from './PickupPointHour'
import Button from './Button'

import styles from './PickupPointDetails.css'
import { LIST } from '../constants'
import { injectState } from '../modalStateContext'

class PickupPointDetails extends Component {
  constructor(props) {
    super(props)

    this.state = {
      unavailableItems: getUnavailableItemsByPickup(
        this.props.items,
        this.props.logisticsInfo,
        this.props.pickupPoint,
        this.props.sellerId
      ),
      items: getItemsWithPickupPoint(
        this.props.items,
        this.props.logisticsInfo,
        this.props.pickupPoint,
        this.props.sellerId
      ),
    }
  }
  handleBackButtonClick = () => {
    this.props.setActiveSidebarState(LIST)
  }

  handleConfirmButtonClick = () => {
    this.props.handleChangeActiveSLAOption({
      slaOption: this.props.pickupPoint.id,
      sellerId: this.props.sellerId,
      shouldUpdateShippingData: false,
    })
    this.props.handleClosePickupPointsModal()
  }

  render() {
    const {
      pickupPoint,
      pickupPointInfo,
      selectedRules,
      isSelectedSla,
      sellerId,
      shouldUseMaps,
      intl,
      storePreferencesData,
      logisticsInfo,
    } = this.props

    const { unavailableItems, items } = this.state

    const businessHours =
      !pickupPointInfo ||
      !pickupPointInfo.businessHours ||
      pickupPointInfo.businessHours.length === 0
        ? null
        : formatBusinessHoursList(pickupPointInfo.businessHours)

    return (
      <div className={`${styles.modalDetails} pkpmodal-details`}>
        <div className={`${styles.modalDetailsTop} pkpmodal-details-top`}>
          <button
            className={`${
              styles.modalDetailsBackLnk
            } pkpmodal-details-back-lnk btn btn-link`}
            onClick={this.handleBackButtonClick}
            type="button">
            <i
              className={`${
                styles.iconBackPickupPointsList
              } pkpmodal-icon-back-pickup-points-list icon-angle-left`}
            />
            {translate(intl, 'cancelBackList')}
          </button>
        </div>

        <div className={`${styles.modalDetailsMiddle} pkpmodal-details-middle`}>
          <div className={`${styles.modalDetailsStore} pkpmodal-details-store`}>
            <PickupPoint
              shouldUseMaps={shouldUseMaps}
              isSelected={isSelectedSla}
              items={this.props.items}
              logisticsInfo={logisticsInfo}
              pickupPoint={pickupPoint}
              selectedRules={selectedRules}
              sellerId={sellerId}
              storePreferencesData={storePreferencesData}
            />
          </div>

          <div className={`${styles.modalDetailsInfo} pkpmodal-details-info`}>
            <div
              className={`${styles.modalDetailsGroup} pkpmodal-details-group`}>
              <h3
                className={`${
                  styles.modalDetailsInfoTitle
                } title pkpmodal-details-info-title`}>
                {translate(intl, 'productsInPoint')}
              </h3>
              {items && <ProductItems items={items} />}
              {unavailableItems && (
                <ProductItems isAvailable={false} items={unavailableItems} />
              )}
            </div>
            {pickupPoint.pickupStoreInfo &&
              pickupPoint.pickupStoreInfo.additionalInfo && (
                <div
                  className={`${
                    styles.modalDetailsGroup
                  } pkpmodal-details-group`}>
                  <h3
                    className={`${
                      styles.modalDetailsInfoTitle
                    } pkpmodal-details-info-title`}>
                    {translate(intl, 'aditionalInfo')}
                  </h3>
                  {pickupPoint.pickupStoreInfo.additionalInfo}
                </div>
              )}

            {businessHours && (
              <div
                className={`${
                  styles.modalDetailsGroup
                } pkpmodal-details-group`}>
                <h3
                  className={`${
                    styles.modalDetailsInfoTitle
                  } pkpmodal-details-info-title`}>
                  {translate(intl, 'businessHours')}
                </h3>
                <table
                  className={`${
                    styles.modalDetailsHours
                  } pkpmodal-details-hours`}>
                  <tbody>
                    {businessHours.map((day, i) => {
                      return (
                        <tr key={i}>
                          <td
                            className={`${
                              styles.modalDetailsHoursDay
                            } pkpmodal-details-hours-day`}>
                            {translate(intl, `weekDay${day.number}`)}
                          </td>
                          {day.closed ? (
                            <td
                              className={`${
                                styles.modalDetailsHoursClosed
                              } pkpmodal-details-hours-closed`}>
                              {translate(intl, 'closed')}
                            </td>
                          ) : (
                            <td
                              className={`${
                                styles.modalDetailsHoursRange
                              } pkpmodal-details-hours-range`}>
                              <PickupPointHour time={day.openingTime} />{' '}
                              {translate(intl, 'hourTo')}{' '}
                              <PickupPointHour time={day.closingTime} />
                            </td>
                          )}
                        </tr>
                      )
                    })}
                  </tbody>
                </table>
              </div>
            )}
          </div>
        </div>

        <div className={`${styles.modalDetailsBottom} pkpmodal-details-bottom`}>
          <Button
            id={`confirm-pickup-${pickupPoint.id
              .replace(/[^\w\s]/gi, '')
              .split(' ')
              .join('-')}`}
            kind="primary"
            large
            moreClassName={`${
              styles.modalDetailConfirmBtn
            } pkpmodal-details-confirm-btn`}
            onClick={this.handleConfirmButtonClick}
            title={translate(intl, 'confirmPoint')}
          />
        </div>
      </div>
    )
  }
}

PickupPointDetails.propTypes = {
  handleChangeActiveSLAOption: PropTypes.func.isRequired,
  handleClosePickupPointsModal: PropTypes.func.isRequired,
  intl: intlShape,
  isSelectedSla: PropTypes.bool,
  items: PropTypes.array.isRequired,
  logisticsInfo: PropTypes.array.isRequired,
  onClickPickupModal: PropTypes.func,
  pickupPoint: PropTypes.object.isRequired,
  pickupPointInfo: PropTypes.object.isRequired,
  selectedRules: PropTypes.object.isRequired,
  sellerId: PropTypes.string,
  storePreferencesData: PropTypes.object.isRequired,
}

export default injectState(injectIntl(PickupPointDetails))
