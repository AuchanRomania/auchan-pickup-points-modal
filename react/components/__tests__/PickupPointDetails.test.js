import React from 'react'
import PickupPointDetails from '../PickupPointDetails'

import { mount } from 'enzyme'
import { render, fireEvent } from 'react-testing-library'
import { Provider } from 'react-redux'
import { IntlProvider } from 'react-intl'
import renderer from 'react-test-renderer'
import { addValidation } from '@vtex/address-form'
import BRA from '@vtex/address-form/lib/country/BRA'
import { PICKUP, DELIVERY, PICKUP_IN_STORE, SIDEBAR } from '../../constants'
import messages from '../../../messages/en.json'
import { ModalStateContext } from '../../modalStateContext'
import ModalState from '../../ModalState'

jest.mock('../../utils/Images', () => ({
  fixImageUrl: () => 'teste.png',
}))
describe('PickupPointDetails', () => {
  let state,
    store,
    props,
    handleChangeActiveSLAOption,
    handleClosePickupPointsModal,
    togglePickupDetails,
    selectNextPickupPoint,
    setActiveSidebarState,
    modalState

  const address = {
    addressType: 'residential',
    receiverName: null,
    addressId: '10',
    postalCode: '22222222',
    city: 'Rio de janeiro',
    state: 'RJ',
    country: 'BRA',
    street: 'Av das Américas',
    number: '',
    neighborhood: 'Barra da Tijuca',
    complement: 'Loja Barra da Tijuca',
    reference: null,
    geoCoordinates: { value: [] },
    addressQuery: 'query',
  }

  beforeEach(() => {
    handleChangeActiveSLAOption = jest.fn()
    handleClosePickupPointsModal = jest.fn()
    togglePickupDetails = jest.fn()
    setActiveSidebarState = jest.fn()
    selectNextPickupPoint = jest.fn()

    modalState = {
      activeState: SIDEBAR,
      shouldUseMaps: true,
      setActiveSidebarState,
      shouldUseMaps: false,
      setShouldSearchArea: jest.fn(),
      setSelectedPickupPoint: jest.fn(),
      selectNextPickupPoint,
      pickupPoints: [
        {
          id: '1',
          pickupPointId: '1',
          friendlyName: 'Loja VTEX',
          name: 'test',
          address: { geoCoordinates: [123, 123] },
          businessHours: [
            { DayOfWeek: 0, OpeningTime: '02:00:00', ClosingTime: '14:00:00' },
            { DayOfWeek: 6, OpeningTime: '03:00:00', ClosingTime: '11:30:00' },
          ],
        },
        {
          id: '2',
          pickupPointId: '2',
          friendlyName: 'Outra Loja',
          name: 'test2',
          address: { geoCoordinates: [125, 125] },
          businessHours: [
            { ClosingTime: '12:00:00', DayOfWeek: 0, OpeningTime: '02:00:00' },
            { ClosingTime: '22:30:00', DayOfWeek: 6, OpeningTime: '16:00:00' },
          ],
        },
      ],
      pickupOptions: [
        {
          id: '1',
          pickupPointId: '1',
          friendlyName: 'Loja VTEX',
          address: { geoCoordinates: [123, 123] },
          pickupStoreInfo: {
            isPickupStore: true,
            friendlyName: 'Loja VTEX',
            address: {
              addressType: 'pickup',
              receiverName: null,
              addressId: '1f848d7',
              isDisposable: true,
              postalCode: '22230061',
              city: 'Rio de Janeiro',
              state: 'RJ',
              country: 'BRA',
              street: 'Rua Marquês de Abrantes',
              number: '5',
              neighborhood: 'Flamengo',
              complement: '',
              reference: null,
              geoCoordinates: [-43.17771, -22.93772],
            },
            additionalInfo: 'não entre de sunga\nnão entre sem sunga',
            dockId: '11dbf12',
          },
        },
        {
          id: '2',
          pickupPointId: '2',
          friendlyName: 'Outra Loja',
          address: { geoCoordinates: [125, 125] },
          pickupStoreInfo: {
            isPickupStore: true,
            friendlyName: 'Outra Loja',
            address: {
              addressType: 'pickup',
              receiverName: null,
              addressId: '1f848d7',
              isDisposable: true,
              postalCode: '22230061',
              city: 'Rio de Janeiro',
              state: 'RJ',
              country: 'BRA',
              street: 'Rua Marquês de Abrantes',
              number: '5',
              neighborhood: 'Flamengo',
              complement: '',
              reference: null,
              geoCoordinates: [-43.17771, -22.93772],
            },
            additionalInfo: 'não entre de sunga\nnão entre sem sunga',
            dockId: '11dbf12',
          },
        },
      ],
      selectedPickupPoint: {
        name: 'Loja VTEX',
        price: 100,
        shippingEstimate: '1bd',
        pickupStoreInfo: {
          friendlyName: 'Loja VTEX',
          address: { geoCoordinates: [123, 123] },
        },
        deliveryChannel: PICKUP_IN_STORE,
        id: '1',
        pickupPointId: '1',
      },
      logisticsInfo: [
        {
          itemIndex: 0,
          deliveryChannels: [{ id: DELIVERY }],
          selectedSla: '2',
          slas: [
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '1',
              pickupStoreInfo: { friendlyName: 'test', address },
            },
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '2',
              pickupStoreInfo: { friendlyName: 'test', address },
            },
          ],
        },
        {
          itemIndex: 1,
          deliveryChannels: [{ id: PICKUP_IN_STORE }],
          selectedSla: '2',
          slas: [
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '1',
              pickupStoreInfo: { friendlyName: 'test', address },
            },
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '2',
              pickupStoreInfo: { friendlyName: 'test', address },
            },
          ],
        },
        {
          itemIndex: 2,
          deliveryChannels: [{ id: PICKUP_IN_STORE }],
          selectedSla: '2',
          slas: [
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '1',
              pickupStoreInfo: { friendlyName: 'test', address },
            },
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '3',
              pickupStoreInfo: { friendlyName: 'test', address },
            },
          ],
        },
      ],
    }

    state = {
      pickup: {
        activeSellerId: '1',
        pickupPointId: '2',
        pickupOptions: [
          {
            name: 'test',
            price: 100,
            shippingEstimate: '1bd',
            pickupStoreInfo: {
              friendlyName: 'test',
              address: {
                geoCoordinates: [123, 123],
              },
            },
            deliveryChannel: PICKUP_IN_STORE,
            id: '1',
          },
          {
            name: 'test',
            price: 100,
            shippingEstimate: '1bd',
            pickupStoreInfo: {
              friendlyName: 'test',
              address: {
                geoCoordinates: [123, 123],
              },
            },
            deliveryChannel: PICKUP_IN_STORE,
            id: '2',
          },
        ],
      },
      orderForm: {
        googleMapsKey: '1234',
        clientPreferencesData: {
          locale: 'pt-BR',
        },
        items: [
          {
            name: 'test',
            imageUrl: 'test.png',
            uniqueId: 'xablau',
            seller: '1',
          },
          {
            name: 'test2',
            imageUrl: 'test2.png',
            uniqueId: 'xablau2',
            seller: '1',
          },
          {
            name: 'test3',
            imageUrl: 'test2.png',
            uniqueId: 'xablau3',
            seller: '2',
          },
        ],
        activeTab: PICKUP,
        storePreferencesData: {
          countryCode: 'BRA',
          currencyCode: 'BRL',
          currencySymbol: 'R$',
          currencyFormatInfo: {
            currencyDecimalDigits: 2,
            currencyDecimalSeparator: ',',
            currencyGroupSeparator: '.',
            currencyGroupSize: 3,
            startsWithCurrencySymbol: true,
          },
        },
        shippingData: {
          address: addValidation({
            addressType: 'residential',
            receiverName: null,
            addressId: '10',
            postalCode: '22222222',
            city: 'Rio de janeiro',
            state: 'RJ',
            country: 'BRA',
            street: 'Av das Américas',
            number: '',
            neighborhood: 'Barra da Tijuca',
            complement: 'Loja Barra da Tijuca',
            reference: null,
            geoCoordinates: [],
            addressQuery: 'query',
          }),
          logisticsInfo: [
            {
              itemIndex: 0,
              deliveryChannels: [{ id: DELIVERY }],
              selectedSla: '2',
              slas: [
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '1',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '2',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
              ],
            },
            {
              itemIndex: 1,
              deliveryChannels: [{ id: PICKUP_IN_STORE }],
              selectedSla: '2',
              slas: [
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '1',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
                {
                  name: 'test2',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '2',
                  pickupStoreInfo: {
                    friendlyName: 'test2',
                    address,
                  },
                },
              ],
            },
            {
              itemIndex: 2,
              deliveryChannels: [{ id: PICKUP_IN_STORE }],
              selectedSla: '2',
              slas: [
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '1',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
                {
                  name: 'test3',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '3',
                  pickupStoreInfo: {
                    friendlyName: 'test3',
                    address,
                  },
                },
              ],
            },
          ],
        },
      },
    }
    store = {
      subscribe: jest.fn(),
      dispatch: jest.fn(),
      getState: jest.fn(() => state),
    }

    props = {
      sellerId: '1',
      items: [
        {
          seller: '1',
          name: 'test',
          imageUrl: 'test.png',
          uniqueId: 'xablau',
        },
        {
          seller: '1',
          name: 'test2',
          imageUrl: 'test2.png',
          uniqueId: 'xablau2',
        },
      ],
      unavailableItems: [
        {
          name: 'test3',
          imageUrl: 'test2.png',
          uniqueId: 'xablau3',
        },
      ],
      isSelectedSla: false,
      unavailableItemsAmount: 1,
      storePreferencesData: {
        countryCode: 'BRA',
        currencyCode: 'BRL',
        currencySymbol: 'R$',
        currencyFormatInfo: {
          currencyDecimalDigits: 2,
          currencyDecimalSeparator: ',',
          currencyGroupSeparator: '.',
          currencyGroupSize: 3,
          startsWithCurrencySymbol: true,
        },
      },
      pickupPointId: '2',
      pickupOptionGeolocations: [[123, 123], [123, 123]],
      googleMapsKey: '1234',
      address: addValidation({
        addressType: 'residential',
        receiverName: null,
        addressId: '10',
        postalCode: '22222222',
        city: 'Rio de janeiro',
        state: 'RJ',
        country: 'BRA',
        street: 'Av das Américas',
        number: '',
        neighborhood: 'Barra da Tijuca',
        complement: 'Loja Barra da Tijuca',
        reference: null,
        geoCoordinates: [],
        addressQuery: 'query',
      }),
      pickupPoint: {
        name: 'test',
        price: 100,
        shippingEstimate: '1bd',
        deliveryChannel: PICKUP_IN_STORE,
        id: '1',
        pickupDistance: 10000,
        pickupStoreInfo: {
          friendlyName: 'test',
          address: {
            addressId: '10',
            addressType: 'residential',
          },
          additionalInfo: 'test aditionalInfo',
        },
      },
      selectedRules: BRA,
      handleChangeActiveSLAOption,
      handleClosePickupPointsModal,
      togglePickupDetails,
      logisticsInfo: [],
      pickupPointInfo: {},
    }
  })

  it('should render self and components', () => {
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <ModalStateContext.Provider value={modalState}>
            <IntlProvider
              locale="pt"
              messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
              <PickupPointDetails {...props} />
            </IntlProvider>
          </ModalStateContext.Provider>
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should simulate go back to list of pickups', () => {
    const wrapper = mount(
      <Provider store={store}>
        <ModalStateContext.Provider value={modalState}>
          <IntlProvider
            locale="pt"
            messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
            <PickupPointDetails {...props} />
          </IntlProvider>
        </ModalStateContext.Provider>
      </Provider>
    )

    const backLink = wrapper.find('button.pkpmodal-details-back-lnk')

    backLink.simulate('click')

    expect(setActiveSidebarState.mock.calls).toHaveLength(1)
  })

  it('should simulate confirm a pickupPoint', () => {
    const wrapper = mount(
      <ModalStateContext.Provider value={modalState}>
        <IntlProvider
          locale="pt"
          messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
          <PickupPointDetails {...props} />
        </IntlProvider>
      </ModalStateContext.Provider>
    )

    const confirmButton = wrapper.find('.pkpmodal-details-confirm-btn')

    confirmButton.simulate('click')

    expect(handleClosePickupPointsModal.mock.calls).toHaveLength(1)
  })

  it('should show the right info about the selected pickup point', () => {
    const { queryByText, getByTestId } = render(
      <ModalState
        activePickupPoint={undefined}
        address={address}
        residentialAddress={undefined}
        askForGeolocation={false}
        googleMapsKey={'1234'}
        isSearching={false}
        items={state.orderForm.items}
        mapStatus={'HIDE_MAP'}
        logisticsInfo={modalState.logisticsInfo}
        pickupPoints={modalState.pickupPoints}
        pickupOptions={modalState.pickupOptions}
        salesChannel={undefined}
        orderFormId={undefined}
        selectedPickupPoint={modalState.selectedPickupPoint}>
        <IntlProvider
          locale="pt"
          messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
          <PickupPointDetails {...props} />
        </IntlProvider>
      </ModalState>
    )

    expect(queryByText('Loja VTEX')).toBeTruthy()
    expect(queryByText('03:00')).toBeTruthy()
    expect(queryByText('11:30')).toBeTruthy()

    const nextPickupPointButton = getByTestId('goToNextPickupPoint')
    const previousPickupPointButton = getByTestId('goToPreviousPickupPoint')

    fireEvent.click(nextPickupPointButton)
    expect(queryByText('Loja VTEX')).toBeFalsy()
    expect(queryByText('03:00')).toBeFalsy()
    expect(queryByText('11:30')).toBeFalsy()

    expect(queryByText('Outra Loja')).toBeTruthy()
    expect(queryByText('16:00')).toBeTruthy()
    expect(queryByText('22:30')).toBeTruthy()

    fireEvent.click(previousPickupPointButton)
    expect(queryByText('Outra Loja')).toBeFalsy()
    expect(queryByText('16:00')).toBeFalsy()
    expect(queryByText('22:30')).toBeFalsy()

    expect(queryByText('Loja VTEX')).toBeTruthy()
    expect(queryByText('03:00')).toBeTruthy()
    expect(queryByText('11:30')).toBeTruthy()
  })
})
