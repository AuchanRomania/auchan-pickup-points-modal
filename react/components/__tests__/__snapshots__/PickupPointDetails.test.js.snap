// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickupPointDetails should render self and components 1`] = `
<div
  className="modalDetails pkpmodal-details"
>
  <div
    className="modalDetailsTop pkpmodal-details-top"
  >
    <button
      className="modalDetailsBackLnk pkpmodal-details-back-lnk btn btn-link"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="backChevron"
        fill="none"
        height="14"
        viewBox="0 0 10 14"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0.75845 7.72356C0.345468 7.32958 0.345467 6.67042 0.758449 6.27644L7.33764 0L9.23999 1.81481L3.8047 7L9.23999 12.1852L7.33764 14L0.75845 7.72356Z"
          fill="#4C8EEF"
        />
      </svg>
      See all pickup points
    </button>
  </div>
  <div
    className="pickupDetailsHeader pkpmodal-details-header"
  >
    <p
      className="pickupDetailsHeaderTitle pkpmodal-details-header-title"
    >
      Pickup point details
    </p>
    <div
      className="pickupDetailsHeaderButtons pkpmodal-details-header-buttons"
    >
      <button
        className="pickupDetailsHeaderButton pkpmodal-details-header-button"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height="11"
          viewBox="0 0 16 11"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0.249779 5.5C0.249779 5.75781 0.333112 5.92969 0.499779 6.10156L5.24978 11L6.41644 9.79687L3.08311 6.35938L15.75 6.35938V4.64063L3.08311 4.64063L6.41644 1.20313L5.24978 0L0.499779 4.89844C0.333112 5.07031 0.249779 5.24219 0.249779 5.5Z"
            fill="#4C8EEF"
          />
        </svg>
      </button>
      <button
        className="pickupDetailsHeaderButton pkpmodal-details-header-button"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height="11"
          viewBox="0 0 16 11"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.7502 5.5C15.7502 5.24219 15.6669 5.07031 15.5002 4.89844L10.7502 9.53674e-07L9.58356 1.20313L12.9169 4.64063L0.25 4.64062L0.25 6.35937L12.9169 6.35938L9.58356 9.79688L10.7502 11L15.5002 6.10156C15.6669 5.92969 15.7502 5.75781 15.7502 5.5Z"
            fill="#4C8EEF"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="modalDetailsMiddle pkpmodal-details-middle"
  >
    <div
      className="modalDetailsStore pkpmodal-details-store"
    >
      <div
        className="pickupPoint pkpmodal-pickup-point"
        id="1"
        onClick={[Function]}
      >
        <div
          className="pickupPointMain pkpmodal-pickup-point-main"
        >
          <div
            className="pickupPointMarkerPostalCode pkpmodal-pickup-point-marker"
          >
            <svg
              className="pkpmodal-pickup-point-marker-image"
              height="32"
              version="1.1"
              viewBox="0 0 48 48"
              width="32"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M24,1.11224c-9.38879,0-17,7.61115-17,17 c0,10.1424,12.87262,23.22955,16.2149,26.4566c0.44031,0.42517,1.12988,0.42517,1.57025,0C28.12744,41.3418,41,28.25464,41,18.11224 C41,8.72339,33.38879,1.11224,24,1.11224z"
                fill="#0176ff"
                stroke="#ffffff"
              />
              <circle
                cx="24"
                cy="18"
                fill="#FFFFFF"
                r="6"
              />
            </svg>
            <svg
              className="pkpmodal-pickup-point-marker-image"
              height="32"
              version="1.1"
              viewBox="0 0 48 48"
              width="32"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M24,1.11224c-9.38879,0-17,7.61115-17,17 c0,10.1424,12.87262,23.22955,16.2149,26.4566c0.44031,0.42517,1.12988,0.42517,1.57025,0C28.12744,41.3418,41,28.25464,41,18.11224 C41,8.72339,33.38879,1.11224,24,1.11224z"
                fill="#0176ff"
                stroke="#ffffff"
              />
              <circle
                cx="24"
                cy="18"
                fill="#FFFFFF"
                r="6"
              />
            </svg>
          </div>
          <div
            className="pickupPointInfo pkpmodal-pickup-point-info"
          >
            <p
              className="pickupPointName pkpmodal-pickup-point-name"
            >
              test
            </p>
            <div
              className="pickupPointAddress pkpmodal-pickup-point-address "
            >
              <div
                className="address-summary address-summary-BRA"
              >
                <br
                  className="line1-delimiter"
                />
                <span
                  className="country"
                >
                  BRA
                </span>
              </div>
            </div>
            <span
              className="pickupPointAllAvailable pkpmodal-pickup-point-available"
            >
              All products are available
            </span>
          </div>
        </div>
        <div
          className="pickupPointSlaAvailabilityPostalCode pkpmodal-pickup-point-sla-availability"
        >
          <span
            className="pickupPointPrice pkpmodal-pickup-point-price"
          >
            R$ 1,00
          </span>
          <span
            className="pickupPointSla pkpmodal-pickup-point-sla"
          >
            TranslateEstimate
          </span>
        </div>
      </div>
    </div>
    <div
      className="modalDetailsInfo pkpmodal-details-info"
    >
      <div
        className="modalDetailsGroup pkpmodal-details-group"
      >
        <h3
          className="modalDetailsInfoTitle title pkpmodal-details-info-title"
        >
          Product availability in this pickup point
        </h3>
        <div
          className="productItems pkpmodal-product-items"
        >
          <span
            className="productItem pkpmodal-product-item "
            data-tip="test"
          >
            <img
              alt="test"
              data-tip="test"
              src="teste.png"
            />
            <div
              className="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
              id={undefined}
            />
          </span>
          <span
            className="productItem pkpmodal-product-item "
            data-tip="test2"
          >
            <img
              alt="test2"
              data-tip="test2"
              src="teste.png"
            />
            <div
              className="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
              id={undefined}
            />
          </span>
        </div>
        <div
          className="productItems pkpmodal-product-items"
        />
      </div>
    </div>
  </div>
  <div
    className="modalDetailsBottom pkpmodal-details-bottom"
  >
    <button
      className="modalDetailConfirmBtn pkpmodal-details-confirm-btn btn pkp-modal-btn primary pkp-modal-btn-primary block pkp-modal-btn-block lg pkp-modal-btn-lg "
      disabled={undefined}
      id="confirm-pickup-1"
      onClick={[Function]}
      type="button"
    >
      Pickup in this point
    </button>
  </div>
</div>
`;
